### ISSUE TITLE
Questions about status of branches and my PRs

### ISSUE THREAD

**Issue by beaudet** [ruven/iipsrv#114]
@ruven As far as I can tell, all of my enhancement branches are in sync with yours except for ICC.

icc: it seems you created the icc branch AFTER merging the PNG to master but before reverting PNG from master whereas I had re-baselined my icc branch AFTER you reverted PNG.  It seems that if you just merge the PNG issue would self-resolve.  It's just the history of those branches that aren't in sync.

My cache, lanczos, png are all synced up with yours.  I don't think there's anything for me to do at this point w.r.t. my PRs, but if there's anything I can do to make them easier to merge, just let me know. 

In general, it seems like having a develop branch would allow changes to be merged in must faster - if there are bugs or performance problems, they can be addressed in develop without effecting a production-stable branch.  Occasionally, you could branch develop into a release candidate and after a QA period, it could then be merged into master. I agree with the concerns outlined in #112.

Soon I'll be submitting one enhancement for oversampling and possibly another to support native "IIIF" URL support. These are very small PRs.  I suppose it's probably best to wait on submitting those until after to determine whether or not to create a develop branch or to continue merging to master. I imagine if you created a develop branch, PRs could just all be submitted against that branch without you having to create new branches to accept them.  If there are conflicts, you could just ask the submitter to re-baseline their branch with develop and to resubmit the PR.

Very soon, I'm going to start using in production a version of IIP that merges all of the PRs I have submitted so far, so these changes will be getting decent exposure to production loads.  I need to do this in order to move forward with the eventual elimination of fixed size / statically pre-generated derivatives.  To do that, I needed to address performance and quality concerns with the currently available options.  The cache, IIIF URL support, and Lanczos filter PRs address those concerns.

Thanks,

Dave

**Comment by ruven** [comment]
Yes I agree about the utility of a develop branch and have just created one. I think it's still worth creating new branches for major new features, but otherwise PR's can be submitted directly to master if they are critical bug fixes or to the develop branch.

For the ICC code, perhaps it would be easier if I just recreated it to avoid the out of sync history. You would have to resubmit your PR to it in that case.

For your new small PR's, just submit directly to develop. For the native URL support, I'm just finishing up some code that will do this in a generic protocol-agnostic way and should do everything you need. So, hold that PR for now.

**Comment by beaudet** [comment]
Sounds great.  No problem with resubmitting ICC PR again.  I can easily re-baseline mine and resubmit.  I'll hold off on the IIIF PR for now and will submit the oversampling PR against the develop branch.

**Comment by beaudet** [comment]
@ruven please let me know when you've recreated the ICC branch and I'll resubmit the PR.

**Comment by ruven** [comment]
OK done
